src/Numerical/BLAS/Single.hs

```diff
@@ -81,14 +83,14 @@ sdot n sx incx sy incy
                        + (sx `V.unsafeIndex` i3)*(sy `V.unsafeIndex` i3)
                        + (sx `V.unsafeIndex` i4)*(sy `V.unsafeIndex` i4)
                in  unrolled c' i'
-   -- O(1) - Compute the starting index of an iterative vector traversal
-   -- given the length of the vector and the iterative step size.
-        firstIndex :: Int -- ^ the iterative step
-                  -> Int -- ^ The index of the first iterative step
-        {-# INLINE firstIndex #-}
-        firstIndex !inc
-            | inc>0     = 0
-            | otherwise = (1-n)*inc
+-- O(1) - Compute the starting index of an iterative vector traversal
```

Comment:
The firstIndex function is used by both `sdot` and `sdsdot`.   I have moved it to the top level to facilitate reuse.